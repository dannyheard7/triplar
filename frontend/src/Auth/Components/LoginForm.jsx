import React from "react";

import FormFieldGroup from "Forms/Components/FormFieldGroup";
import NonFieldErrors from "Forms/Components/NonFieldErrors";

export default class LoginForm extends React.Component {
    render() {
        return (
            <form onSubmit={this.props.onSubmit}>
                <NonFieldErrors errors={this.props.errors.non_field_errors}/>
                <FormFieldGroup errors={this.props.errors.email} label="Email" name="email" type="email"
                                onChange={this.props.onChange} required="true"/>
                <FormFieldGroup errors={this.props.errors.password} label="Password" name="password"
                                type="password" onChange={this.props.onChange} required="true"/>
                <button type="submit" className="btn btn-primary" id="login-submit">Login</button>
            </form>
        );
    }
}