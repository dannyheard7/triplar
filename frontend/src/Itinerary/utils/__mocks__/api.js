const faker = require('faker');

export default {
    curryAPIFunc (apiFunc, id) {return (object) => apiFunc(id, object)},

    searchCities(value) {
        return Promise.resolve({status: 200, data: {data: {cities: [faker.address.city(),  faker.address.city()]}}});
    },

    addLocationToTrip(tripId, value) {
        return Promise.resolve({status: 200, data: value});
    }
}