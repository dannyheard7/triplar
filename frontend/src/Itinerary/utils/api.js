import axios from "axios";
import Q from "q";
import {backendHost} from 'App/utils/constants';

export default {
    curryAPIFunc (apiFunc, id) {return (object) => apiFunc(id, object)},

    addLocationToTrip(tripId, destination) {
        const mutateTripLocationQuery = ` 
            mutation AddTripLocation($input: TripLocationMutationInput!){
              result: addTripLocation(input: $input) {
                tripLocation {
                    id
                    startDate
                    endDate
                    city {
                        name
                        location
                    }
                    trip {
                        id
                    }
                }
                errors {
                  field
                  messages
                }
              }
            }`;


        let variables = {input: {trip: tripId, ...destination}};

        return Q.when(axios.post(`${backendHost}/api/graphql`, {query: mutateTripLocationQuery, variables: variables}));
    },

    getLocationDayItinerary(tripLocationId, date) {
        const getLocationDayItinerary = ` 
           query { 
            locationDayItinerary(date: "${date}", locationId: ${tripLocationId}) {
                place {
                    id
                    name
                    imageUrl
                }
            }
           }`;

        return Q.when(axios.post(`${backendHost}/api/graphql`, {query: getLocationDayItinerary}));
    },

    addItemToTripLocation(tripLocationId, placeId, date, order) {
        const addTripLocationItemMutation = ` 
            mutation AddTripLocationItem($input: AddTripLocationItemMutationInput!){
              addTripLocationItem(input: $input) {
                tripLocationItem {
                  order
                }
                errors {
                  field
                  messages
                }
              }
            }`;

        let variables = {input: {location: tripLocationId, apiPlace: placeId, order: order, startDate: date, endDate: date}};

        return Q.when(axios.post(`${backendHost}/api/graphql`, {query: addTripLocationItemMutation, variables: variables}));
    },

    searchCities(query) {
        const searchCities = ` 
           query { 
            cities(name: "${query}") {
                name
                country {
                    name
                }
            }
           }`;

        return Q.when(axios.post(`${backendHost}/api/graphql`, {query: searchCities}));
    },
}