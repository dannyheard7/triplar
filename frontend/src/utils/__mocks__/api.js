const faker = require('faker');

export default {

    getTrips() {
        return new Promise((resolve, reject) => {
            let trips = [faker.random.word(), faker.random.word()];
            let response = {status: 200, data: trips};
            resolve(response);
        })
    },

    createTrip(trip) {
        return new Promise((resolve, reject) => {
            let response = {status: 201, data: trip};
            resolve(response)
        })
    },

    getTrip(id)  {
         const trip = {
            'id': id, 'name': faker.random.word(), 'created_by': faker.internet.email()
        };
        return new Promise((resolve, reject) => {
            let response = {status: 200, data: trip};
            resolve(response);
        });
    }
}