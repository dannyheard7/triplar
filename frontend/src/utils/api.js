import axios from "axios";
import Q from "q";

export default {
    getTrips() {
        return Q.when(axios.get("/trips/"));
    },

    createTrip(trip) {
        return Q.when(axios.post("/trips/", trip));
    },

    getTrip(tripId) {
        return Q.when(axios.get("/trips/" + tripId));
    },

    patchTrip(trip) {
        return Q.when(axios.patch("/trips/" + trip.id, trip));
    },

    deleteTrip(tripId) {
        return Q.when(axios.delete("/trips/" + tripId));
    }
}