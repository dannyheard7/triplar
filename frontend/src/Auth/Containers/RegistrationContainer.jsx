import React from "react";

import RegistrationForm from "Auth/Components/RegistrationForm";
import api from "Auth/utils/auth.api";
import FormContainer from "Forms/Containers/FormContainer";
import {Helmet} from "react-helmet";

export default class RegistrationContainer extends React.Component {

    constructor(props) {
        super(props);

        this.onSuccess = this.onSuccess.bind(this);
    }

    onSuccess(data) {
        this.props.history.push('/login')
    }

    render() {
        return (
            <div>
                <Helmet>
                    <title>Register | Triplar</title>
                </Helmet>
                <FormContainer onSuccess={this.onSuccess} apiFunction={api.registerUser}>
                    <RegistrationForm />
                </FormContainer>
            </div>
        );
    }
}
