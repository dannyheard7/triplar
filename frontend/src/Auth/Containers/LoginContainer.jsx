import React from "react";
import LoginForm from "../Components/LoginForm.jsx";
import {facebookLoginRequest, loginRequest} from "../utils/actions";
import {connect} from "react-redux";
import ReduxFormContainer from "../../Forms/Containers/ReduxFormContainer";
import {withRouter} from "react-router";
import {Helmet} from "react-helmet";
import FacebookLogin from 'react-facebook-login';

export class LoginContainer extends React.Component {

    facebookResponse = (response) => {
        this.props.dispatch(facebookLoginRequest(response.accessToken));
    };


    render() {
        return (
            <div>
                <Helmet>
                    <title>Login | Triplar</title>
                </Helmet>
                <h1>Login</h1>

                {this.props.auth.requesting ?
                    <p>Attempting to login...</p> :
                    <ReduxFormContainer action={loginRequest}>
                        <LoginForm/>
                    </ReduxFormContainer>
                }

                <FacebookLogin
                    appId="2240458192891426"
                    fields="name,email"
                    callback={this.facebookResponse} />
            </div>
        );
    }
}

const mapStateToProps = state => ({
  auth: state.auth,
});

// make Redux state piece of `login` and our action `loginRequest`
// available in this.props within our component
export default withRouter(connect(mapStateToProps)(LoginContainer))