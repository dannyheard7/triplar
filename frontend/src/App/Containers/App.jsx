import React from "react";
import {Route, Router, Switch} from "react-router-dom";
import {DragDropContext} from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';
import {connect} from "react-redux";
import ReactGA from 'react-ga';
import {Helmet} from "react-helmet";

import {history} from "../../store";
import {trackingId} from "../utils/constants";

import LoggedInContainer from "App/Containers/LoggedInContainer";
import HeaderAuthenticated from "App/Components/HeaderAuthenticated";
import HeaderUnauthenticated from "App/Components/HeaderUnauthenticated";
import Footer from "App/Components/Footer";
import LoginContainer from "Auth/Containers/LoginContainer";
import RegistrationContainer from "Auth/Containers/RegistrationContainer";
import TripCreateContainer from "Trips/Components/TripCreate";
import TripEditContainer from "../../Trips/Components/TripEdit";
import TripDelete from "../../Trips/Components/TripDelete";
import PlaceDetail from "../../Places/Components/PlaceDetail";
import {getCookie} from "../utils/utils";
import {tokenRefresh} from "../../Auth/utils/actions";
import TripDetail from "../../Trips/Components/TripDetail";
import TripList from "../../Trips/Components/TripList";
import ItinerariesOverview from "../../Itinerary/Components/ItinerariesOverview";
import LocationItinerary from "../../Itinerary/Components/LocationItinerary";

import 'App/styles/app.css';

export class App extends React.Component {
    componentDidMount() {
        if(!this.props.auth.successful) {
            const loginToken = getCookie('userToken');

            if (loginToken) {
                this.props.dispatch(tokenRefresh(loginToken));
            }
        }

        this.initializeReactGA();
    }

    initializeReactGA() {
        ReactGA.initialize('UA-130249377-1');
        ReactGA.pageview("/");
        history.listen(location => ReactGA.pageview(location.pathname));
    }

    render() {
        if(this.props.auth.successful) {
            return (
                <Router history={history}>
                    <LoggedInContainer token={this.props.auth.token}>
                        <Helmet>
                            <title>Triplar</title>
                        </Helmet>
                        <div className="wrapper">
                            <HeaderAuthenticated/>
                            <div className="content">
                                <Route path="/(|trips)" exact component={TripList}/>
                                <Route path="/(|trips)" exact component={TripCreateContainer}/>

                                <Route path="/trips/:id" component={TripDetail}/>
                                <Route path="/trips/:id/edit" component={TripEditContainer}/>
                                <Route path="/trips/:id/delete" component={TripDelete} />

                                <Route path="/trips/:id" component={ItinerariesOverview}/>

                                <Route path="/trips/:tripId/itinerary/:itineraryId" component={LocationItinerary}/>

                                <Route path="*/place/:placeId" component={PlaceDetail}/>
                            </div>
                            <Footer/>
                        </div>
                    </LoggedInContainer>
                </Router>
            );
        } else {
            return (
                <Router history={history}>
                    <div className="wrapper">
                        <Helmet>
                            <title>Triplar</title>
                        </Helmet>
                        <HeaderUnauthenticated/>
                        <div className="content">
                            <Switch>
                                <Route path="/register" exact component={RegistrationContainer}/>
                                <Route path="/" component={LoginContainer}/>
                            </Switch>
                        </div>
                        <Footer/>
                    </div>
                </Router>
            );
        }
    }
}

const mapStateToProps = state => {
    if(state.auth.successful) {
        return {
            auth: {...state.auth, token: getCookie('userToken')}
        }
    } else {
        return {auth: state.auth}
    }
};

const connectedApp = connect(mapStateToProps)(App);

export default DragDropContext(HTML5Backend)(connectedApp)