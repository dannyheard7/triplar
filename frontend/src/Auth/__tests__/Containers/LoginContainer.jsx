import React from "react";
import {shallow} from "enzyme";
import api from "utils/api.js";
import ShallowRenderer from 'react-test-renderer/shallow';

import {LoginContainer} from "Auth/Containers/LoginContainer";
import LoginForm from "Auth/Components/LoginForm";

const actions = require('Auth/actions');
const renderer = new ShallowRenderer();
const faker = require('faker');

jest.mock('utils/api.js');

describe('<LoginContainer />', () => {
    const event = {
        preventDefault: () => {},
        target: {
            value: "",
            name: "",
        },
    };

    test('calls handle submit after receiving submit from <LoginForm />', () => {
        const stub = jest.spyOn(LoginContainer.prototype, 'handleSubmit');

        const container = shallow(<LoginContainer />);
        container.find(LoginForm).prop('onSubmit')(event);

        expect(stub).toBeCalled();
    });

    test('calls handle change after receiving change from <LoginForm />', () => {
        const stub = jest.spyOn(LoginContainer.prototype, 'handleChange');

        const container = shallow(<LoginContainer />);
        container.find(LoginForm).prop('onChange')(event);

        expect(stub).toBeCalled();
    });

    test('renders correctly', () => {
        const result = renderer.render(<LoginContainer  />);
        expect(result).toMatchSnapshot();
    });

    test('updates the state when handleChange is called', () => {
        const container = shallow(<LoginContainer />);
        const instance = container.instance();
        const email = faker.internet.email();

        instance.handleChange({'target': {'name': 'email', 'value': email}});
        expect(container.state('email')).toEqual(email);
    });

    test('errors state change updates LoginForm Prop', () => {
        const container = shallow(<LoginContainer />);
        expect(container.find(LoginForm).props().errors).toEqual(container.state('errors'));

        let errors = [faker.lorem.word, faker.lorem.word];
        container.setState({errors: errors});

        expect(container.find(LoginForm).props().errors).toEqual(errors);
    });

    // Need to convert to new mocks
    describe('api', () => {
        const loginStub = jest.spyOn(actions, 'login');
        const routerSpy = {
            push: jest.fn()
        };
        const dispatchStub = jest.fn();

        afterEach(function () {
            loginStub.mockReset();
            routerSpy.push.mockReset();
            dispatchStub.mockReset();
        });

        test('succesful registration dispatches login', async () => {
            const container = shallow(<LoginContainer router={routerSpy} dispatch={dispatchStub}/>);

            const loginStubReturn = jest.fn();
            loginStub.mockReturnValue(loginStubReturn);

            container.instance().login();

            await Promise.resolve();

            expect(loginStub).toBeCalled();
            expect(dispatchStub).toBeCalledWith(loginStubReturn);
        });

        test('unsuccesful registration updates state errors', async () => {
            const container = shallow(<LoginContainer />);

            const errors = [faker.lorem.word, faker.lorem.word];
            api.getLoginToken = jest.fn().mockReturnValueOnce(new Promise((resolve, reject) => {
                let response = {status: 400, data: errors};
                resolve(response);
            }));

            container.instance().login();
            await Promise.resolve();

            expect(container.state('errors')).toEqual(errors);
            expect(loginStub).not.toBeCalled();
        });
    });
});