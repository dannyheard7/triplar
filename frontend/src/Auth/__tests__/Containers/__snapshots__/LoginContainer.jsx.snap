// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginContainer /> renders correctly 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Login | Triplar
    </title>
  </HelmetWrapper>
  <h1>
    Login
  </h1>
  <Connect(ReduxFormContainer)
    action={[MockFunction]}
  >
    <LoginForm />
  </Connect(ReduxFormContainer)>
</div>
`;
